web/react/utils/markdown.jsx

```diff
@@ -39,7 +40,14 @@ export class MattermostMarkdownRenderer extends marked.Renderer {
         if (title) {
             output += ' title="' + title + '"';
         }
-        output += ' target="_blank">' + text + '</a>';
+
+        if (outHref.indexOf(Utils.getTeamURLFromAddressBar()) === 0) {
```

Comment:
You also need to make this change in the autolinkUrls function in web/react/utils/text_formatting.jsx since that handles the autolinking of most links. This block just does markdown links specified like `[text](link)`.

Also, we might want to switch this to lastIndexOf like where we check for http above since it's a bit faster.